src/formatDate.js

```diff
@@ -8,7 +8,48 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const [, , , fromSeparator] = fromFormat;
+  const [, , , toSeparator] = toFormat;
+
+  const dateArray = date.split(fromSeparator);
+  const modifiedDate = [];
+
+  const fromFormatLength = fromFormat.join('').length;
+  const toFormatLength = toFormat.join('').length;
+
+  if (fromFormatLength > toFormatLength) {
+    const fromYearIndex = fromFormat.indexOf('YYYY');
+    const splitArrayDate = dateArray[fromYearIndex].split('');
+
+    for (let i = 1; i < 3; i++) {
+      splitArrayDate.shift();
+    }
+
+    dateArray[fromYearIndex] = splitArrayDate.join('');
+
+    fromFormat[fromYearIndex] = 'YY';
+  } else if (fromFormatLength < toFormatLength) {
+    const fromYearIndex = fromFormat.indexOf('YY');
+    const splitArrayDate = dateArray[fromYearIndex].split('');
+
+    if (dateArray[fromYearIndex] < 30) {
+      splitArrayDate.unshift(20);
+    } else {
+      splitArrayDate.unshift(19);
+    }
+
+    dateArray[fromYearIndex] = splitArrayDate.join('');
+
+    fromFormat[fromYearIndex] = 'YYYY';
+  }
+
+  for (let i = 0; i < dateArray.length; i++) {
+    const position = fromFormat.indexOf(toFormat[i]);
```

Comment:
The `indexOf` method will return -1 if the `toFormat[i]` is not found in `fromFormat`, which will cause an error when accessing `dateArray[position]`. Consider adding a check to ensure `position` is valid before using it.